### PR TITLE
Remove pre-configured Okta & Google OAuth providers

### PR DESCRIPTION
**Pull request by akrousset**
Fixes firezone/product#391

### Summary 
- Removed Okta & Google pre-configured providers from codebase & docs.
- Updated `docs/administer/upgrade.md` to include instructions on how to migrate from these to OIDC providers.
- Added a `configCheck` function in the `omnibus/package-scripts/preinst` script to detect enabled out of date providers.
  - Provides a URL to migration instructions in the upgrade document.

### CODE REVIEW

File: omnibus/package-scripts/firezone/preinst [hunk 1]
```diff
@@ -17,6 +17,18 @@ kernelCheck() {
...
 }
 
+configCheck() {
+    config_file='/etc/firezone/firezone.rb'
+    error_color='\033[0;31m'
+    reset_color='\033[0m'
+
+    if grep -q "^\s*default\['firezone'\]\['authentication'\]\['\(google\|okta\)'\]\['enabled'\]\s*=\s*true" $config_file ; then
+        echo "${error_color}ERROR: Firezone 0.5+ disables support for pre-configured Okta and Google OAuth2 providers!${reset_color}"
+        echo "Please see our transition guide to move to a generic OIDC config: https://docs.firezone.dev/administer/upgrade#upgrading-from--050-to-050"
```
Comment (akrousset) [inline omnibus/package-scripts/firezone/preinst]: I believe this is the correct URL once the new docs will be hosted here... I don't believe there's a `docs` directory before the page like the old docs have for <= 0.4.5.

File: docs/docs/administer/upgrade.md [hunk 2]
```diff
@@ -19,6 +19,41 @@ To upgrade Firezone, follow these steps:
...
+### I have an existing Okta OAuth configuration
+
+Remove these lines containing the old Okta OAuth configs from your configuration
+file located at `/etc/firezone/firezone.rb`
+
+```rb
+default['firezone']['authentication']['okta']['enabled']
+default['firezone']['authentication']['okta']['client_id']
+default['firezone']['authentication']['okta']['client_secret']
+default['firezone']['authentication']['okta']['redirect_uri']
```
Comment (jamilbk) [inline docs/docs/administer/upgrade.md]: I think this key was named `default['firezone']['authentication']['okta']['site']`

File: docs/docs/administer/upgrade.md [hunk 3]
```diff
@@ -19,6 +19,41 @@ To upgrade Firezone, follow these steps:
 Occasionally problems arise. If you hit any, please let us know by [filing an
 issue](https://github.com/firezone/firezone/issues/new/choose).
 
+## Upgrading from < 0.5.0 to 0.5.0
```
Comment (jamilbk) [inline docs/docs/administer/upgrade.md]: Probably want a slight clarification to users going from 0.4.x to 0.5.1 and up:

```suggestion
## Upgrading from < 0.5.0 to >= 0.5.0
```

**Review by jamilbk — CHANGES_REQUESTED** [review]
A couple minor clarifications but otherwise LGTM

**Comment by princemaple** [comment]
Do we still need this? https://github.com/firezone/firezone/blob/391-remove-pre-configured-oidc-providers/apps/fz_http/lib/fz_http_web/user_from_auth.ex#L20-L26

File: docs/docs/administer/upgrade.md (hunk 2 above)
Comment (akrousset) [reply docs/docs/administer/upgrade.md]: Yep, it was!

**Comment by akrousset** [comment]
> Do we still need this? https://github.com/firezone/firezone/blob/391-remove-pre-configured-oidc-providers/apps/fz_http/lib/fz_http_web/user_from_auth.ex#L20-L26

Nope, thanks for catching this!

File: omnibus/package-scripts/firezone/preinst (hunk 1 above)
Comment (jamilbk) [reply omnibus/package-scripts/firezone/preinst]: Cool -- this URL may change based on the new subtitle I suggested, probably to

```suggestion
        echo "Please see our transition guide to move to a generic OIDC config: https://docs.firezone.dev/administer/upgrade#upgrading-from--050-to--050"
```

File: omnibus/package-scripts/firezone/preinst [hunk 4]
```diff
@@ -17,6 +17,18 @@ kernelCheck() {
   fi
 }
 
+configCheck() {
+    config_file='/etc/firezone/firezone.rb'
+    error_color='\033[0;31m'
```
Comment (jamilbk) [inline omnibus/package-scripts/firezone/preinst:22]: 👍

**Review by princemaple — APPROVED** [review]

File: omnibus/package-scripts/firezone/preinst (hunk 1 above)
Comment (akrousset) [reply omnibus/package-scripts/firezone/preinst]: Ah yeah sorry, totally skipped my mind. I fixed it 👍

**Review by jamilbk — APPROVED** [review]

**Pull request merged**